src/main/java/uk/gov/pay/directdebit/mandate/model/subtype/MandateExternalId.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.pay.directdebit.mandate.model.subtype;
+
+import java.util.Objects;
+
+public class MandateExternalId {
+
+    private final String mandateExternalId;
+
+    private MandateExternalId(String mandateExternalId) {
+        this.mandateExternalId = Objects.requireNonNull(mandateExternalId);
+    }
+
+    public static MandateExternalId of(String mandateExternalId) {
+        return new MandateExternalId(mandateExternalId);
+    }
+
+    public static MandateExternalId valueOf(String mandateExternalId) {
```

Comment:
Do we need `valueOf` in addition to the `of` method, I don't see it used anywhere?